tensorflow/python/lib/io/file_io.py

```diff
@@ -778,6 +779,26 @@ def list_directory_v2(path):
       for filename in _pywrap_file_io.GetChildren(compat.path_to_bytes(path))
   ]
 
+@tf_export("io.gfile.join")
+def join(path, *paths):
+  """Join one or more path components intelligently.
+
+  This is the same as os.path.join except that it handles
+  TensorFlow specific filesystems like `ram://` and `gcs://`.
+
+  Args:
```

Comment:
> Maybe add examples here of the TF filesystem behavior?

(from internal review)

Reply:
I added an example in the docstring.

Reply:
Could you add a slightly more complex example here? I tried the provided example with os.path.join and I see the same result. So might be helpful to find something that has a different result....

Reply:
results would only differ from os.path.join on Windows

Reply:
On windows you get:

```
>>> tf.io.join("gcs://folder", "file.py") 
"gcs://folder\\file.py"
```

Reply:
I updated the docstring to:
```
This is the same as os.path.join except that it guarantees correct
handling of tensorflow specific filesystems like `gcs://` and `ram://`
on all platforms.
```
To try to clarify. Let me know if that works or if there's any other wording you'd prefer.